components/tools/OmeroJava/test/integration/DuplicationTest.java

```diff
@@ -265,15 +276,31 @@ private static void assertSameProperties(StatsInfo original, StatsInfo duplicate
         Assert.assertEquals(duplicate.getGlobalMax().getValue(), original.getGlobalMax().getValue());
     }
 
+    /**
+     * Assert that the given instances have the same property values.
+     * @param original the original instance
+     * @param duplicate the duplicate instance that is expected to correspond to the original
+     */
+    private static void assertSameProperties(Shape original, Shape duplicate) {
+      Assert.assertEquals(duplicate.getTheZ().getValue(), original.getTheZ().getValue());
+      Assert.assertEquals(duplicate.getTheT().getValue(), original.getTheT().getValue());
+      Assert.assertEquals(duplicate.getTheC().getValue(), original.getTheC().getValue());
+      ExternalInfo originalExternalInfo = original.getDetails().getExternalInfo();
+      Assert.assertNotNull(originalExternalInfo);
+      ExternalInfo duplicateExternalInfo = duplicate.getDetails().getExternalInfo();
+      Assert.assertNotNull(duplicateExternalInfo);
+      Assert.assertEquals(originalExternalInfo.getEntityId(), duplicateExternalInfo.getEntityId());
+      Assert.assertEquals(originalExternalInfo.getEntityType(), duplicateExternalInfo.getEntityType());
+      Assert.assertEquals(originalExternalInfo.getUuid(), duplicateExternalInfo.getUuid());
+    }
+
     /**
      * Assert that the given instances have the same property values.
      * @param original the original instance
      * @param duplicate the duplicate instance that is expected to correspond to the original
      */
     private static void assertSameProperties(Rectangle original, Rectangle duplicate) {
-        Assert.assertEquals(duplicate.getTheZ().getValue(), original.getTheZ().getValue());
-        Assert.assertEquals(duplicate.getTheT().getValue(), original.getTheT().getValue());
-        Assert.assertEquals(duplicate.getTheC().getValue(), original.getTheC().getValue());
+        assertSameProperties((Shape) original, (Shape) duplicate);
```

Comment:
same

Reply:
This is not a call to a testng API but to an internal method. I essentially followed the style of all the existing API 

```
    private static void assertSameProperties(Image original, Image duplicate) {
    private static void assertSameProperties(Pixels original, Pixels duplicate) {
    private static void assertSameProperties(Channel original, Channel duplicate) {
```

Do you want all of these to be updated to list the duplicate object as the first one?